tools/mmp/driver.cs

```diff
@@ -614,6 +614,18 @@ static void Pack (IList<string> unprocessed)
 					else
 						native_libs.Add (kvp.Key, kvp.Value);
 				}
+				// Idealy, this would be handled by Link () above. However in the non-linking case
+				// we do not run MobileMarkStep which generates the pinvoke list. Hack around this for now
+				// https://bugzilla.xamarin.com/show_bug.cgi?id=43419
+				if (App.LinkMode == LinkMode.None) {
+					foreach (var kvp in ProcessDllImports ()) {
```

Comment:
> just do everything in Link

yes, that's why I suggested that the change be made inside `Link`. 

There will always be a difference between non-linked and linked builds. Rolf's previous change just changed the boundary of where it occurred - which is more pain transference, than removal (i.e. now easier, not easy).

Along with the test I'm fine to have this in C8 and to see something cleaner in master (for C9).